command/git_push.go

```diff
@@ -0,0 +1,153 @@
+package command
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/rest/client"
+	restModel "github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip/level"
+	"github.com/pkg/errors"
+)
+
+// gitMerge is a command that commits tracked changes and pushes the resulting commit to a remote repository
+type gitPush struct {
+	// The root directory (locally) that the code is checked out into.
+	// Must be a valid non-blank directory name.
+	Directory string `plugin:"expand"`
+	base
+}
+
+func gitPushFactory() Command   { return &gitPush{} }
+func (c *gitPush) Name() string { return "git.push" }
+
+// ParseParams parses the command's configuration.
+// Fulfills the Command interface.
+func (c *gitPush) ParseParams(params map[string]interface{}) error {
+	err := mapstructure.Decode(params, c)
+	if err != nil {
+		return err
+	}
+
+	if c.Directory == "" {
+		return errors.Errorf("error parsing '%v' params: value for directory "+
+			"must not be blank", c.Name())
+	}
+
+	return nil
+}
+
+// Execute gets the source code required by the project
+func (c *gitPush) Execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *model.TaskConfig) error {
+	p, err := comm.GetTaskPatch(ctx, client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret})
+	if err != nil {
+		return errors.Wrap(err, "Failed to get patch")
+	}
+
+	commands := getCheckoutBranchCommands(conf.ProjectRef.Branch)
```

Comment:
Can we log each git command to the task's logger?

Reply:
done.